internal/controller/cluster/reconciler.go

```diff
@@ -286,36 +286,39 @@ func (r *SingleClusterReconciler) Reconcile() (result ctrl.Result, recErr error)
 	return reconcile.Result{}, nil
 }
 
-func (r *SingleClusterReconciler) recoverIgnorablePods() common.ReconcileResult {
-	podList, gErr := r.getClusterPodList()
-	if gErr != nil {
-		r.Log.Error(gErr, "Failed to get cluster pod list")
-		return common.ReconcileError(gErr)
-	}
-
+func (r *SingleClusterReconciler) recoverIgnorablePods(ignorablePodNames sets.Set[string]) common.ReconcileResult {
 	r.Log.Info("Try to recover failed/pending pods if any")
 
+	ignorablePodNamesList := ignorablePodNames.UnsortedList()
+
 	var anyPodFailed bool
 	// Try to recover failed/pending pods by deleting them
-	for idx := range podList.Items {
-		if cErr := utils.CheckPodFailed(&podList.Items[idx]); cErr != nil {
+	for idx := range ignorablePodNamesList {
+		pod := &corev1.Pod{}
+		podNamespacedName := types.NamespacedName{Namespace: r.aeroCluster.Namespace, Name: ignorablePodNamesList[idx]}
+
+		if err := r.Client.Get(context.TODO(), podNamespacedName, pod); err != nil {
```

Comment:
can we use the older approach of getting the list of all the pods and filter it in code based on ignorablePodNamesList. 
It can reduce the API calls if possible.